### PR TITLE
feat(AbstractTorchFSMolModel): Make model output/loss more flexible

### PR DESCRIPTION
**Pull request by mmjb**

### CODE REVIEW

File: fs_mol/models/abstract_torch_fsmol_model.py [hunk 1]
```diff
@@ -62,10 +74,29 @@ def forward(self, batch: BatchFeaturesType) -> torch.Tensor:
...
+            batch: representation of the features of NUM_MOLECULES, as chosen by the implementor.
+            labels: float Tensor of shape [NUM_MOLECULES], indicating the target label of the each molecule.
+            model_output: output of the model, as chosen by the implementor.
+
+        Returns:
+            Dictionary mapping partial loss names to the loss. Optimization will be performed over the sum of values.
+        """
+        predictions = model_output.molecule_binary_label.squeeze(dim=-1)
+        label_loss = torch.mean(self.criterion(predictions, labels.float()))
+        return {"label_loss": label_loss}
```
[the remark this answers is not in the document]
Comment (megstanley) [reply fs_mol/models/abstract_torch_fsmol_model.py:98]: Sorry, it's probably a waste of time, it is more just for symmetry with model outputs type

[the remark this answers is not in the document]
Comment (megstanley) [reply fs_mol/models/abstract_torch_fsmol_model.py:98]: No, this is fine! I guess I was imagining the case where you already compute the loss but you output components that went in to the  loss for inspection/ you can calculate auxiliary quantities that aren't included in the eventual loss, but I don't think it is important right now!

[the remark this answers is not in the document]
Comment (mmjb) [reply fs_mol/models/abstract_torch_fsmol_model.py]: One step ahead of you: the logger does it on consumption: https://github.com/microsoft/FS-Mol/pull/44/files#diff-aa84a7d6905ef6e7f294079f5b67ff57d13dd9ea35bb42ef03afc2fe9f9fef4fR43
